Explorer/Assets/DCL/SDKComponents/MediaStream/Systems/UpdateMediaPlayerSystem.cs

```diff
@@ -25,17 +26,43 @@ namespace DCL.SDKComponents.MediaStream
     [ThrottlingEnabled]
     public partial class UpdateMediaPlayerSystem : BaseUnityLoopSystem
     {
+        private const string WORLD_VOLUME_DATA_STORE_KEY = "Settings_WorldVolume";
+
         private readonly IWebRequestController webRequestController;
         private readonly ISceneData sceneData;
         private readonly ISceneStateProvider sceneStateProvider;
         private readonly IPerformanceBudget frameTimeBudget;
-
-        public UpdateMediaPlayerSystem(World world, IWebRequestController webRequestController, ISceneData sceneData, ISceneStateProvider sceneStateProvider, IPerformanceBudget frameTimeBudget) : base(world)
+        private readonly WorldVolumeMacBus worldVolumeMacBus;
+        private float volume = 1f;
+
+        public UpdateMediaPlayerSystem(
+            World world,
+            IWebRequestController webRequestController,
+            ISceneData sceneData,
+            ISceneStateProvider sceneStateProvider,
+            IPerformanceBudget frameTimeBudget,
+            WorldVolumeMacBus worldVolumeMacBus) : base(world)
         {
             this.webRequestController = webRequestController;
             this.sceneData = sceneData;
             this.sceneStateProvider = sceneStateProvider;
             this.frameTimeBudget = frameTimeBudget;
+            this.worldVolumeMacBus = worldVolumeMacBus;
+
+            //This following part is a workaround applied for the MacOS platform, the reason
+            //is related to the video and audio streams, the MacOS environment does not support
+            //the volume control for the video and audio streams, as it doesn’t allow to route audio
+            //from HLS through to Unity. This is a limitation of Apple’s AVFoundation framework
+#if UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX
+            this.worldVolumeMacBus.OnWorldVolumeChanged += OnWorldVolumeChanged;
```

Comment:
I would try to keep all dirtyness contained in a single class, `WorldVolumeMacBus`. Put the compilation directives there, as well as adding a `GetVolumePercentage()` function